sql/sql_admin.cc

```diff
@@ -1065,7 +1065,8 @@ bool Sql_cmd_analyze_table::execute(THD *thd)
   if (check_table_access(thd, SELECT_ACL | INSERT_ACL, first_table,
                          FALSE, UINT_MAX, FALSE))
     goto error;
-  thd->enable_slow_log= opt_log_slow_admin_statements;
+  thd->enable_slow_log= opt_log_slow_admin_statements
+	  && (thd->sp_runtime_ctx ? opt_log_slow_sp_statements : 1 );
```

Comment:
Can this and identical assignments below be pulled up to next to lex->m_sql_cmd->execute(thd) call in "case SQLCOM_ANALYZE etc" in mysql_execute_command?